modules/thermal_hydraulics/include/components/Component.h

```diff
@@ -457,6 +457,44 @@ class Component : public THMObject,
                    const std::string & subdomain_name,
                    const Moose::CoordinateSystemType & coord_system = Moose::COORD_XYZ);
 
+  /**
+   * Adds a functor material to compute the absolute nonlinear step in some functor
+   *
+   * @param[in] functor_name  Functor for which to compute step
+   * @param[in] property  Name of new step functor material property
+   * @param[in] functor_is_ad  Is the functor for which to compute step AD?
+   */
+  void addNonlinearStepFunctorMaterial(const std::string & functor_name,
+                                       const std::string & property,
+                                       bool functor_is_ad);
+
+  /**
+   * Adds a Postprocessor to compute the maximum of a functor over some domain
+   *
+   * @param[in] functor_name  Functor for which to compute maximum
+   * @param[in] pp_name  Name of new Postprocessor
+   * @param[in] normalization  Factor by which to divide quantity
+   * @param[in] subdomains  Subdomains over which to compute maximum
+   */
+  void addMaximumFunctorPostprocessor(const std::string & functor_name,
+                                      const std::string & pp_name,
+                                      Real normalization,
+                                      const std::vector<SubdomainName> & subdomains);
+
+  /**
+   * Adds a MultiPostprocessorConvergence for nonlinear convergence for the component
+   *
+   * @param[in] postprocessors  Postprocessors to compare
+   * @param[in] descriptions  Description of each Postprocessor
+   * @param[in] tolerances  Tolerance for each check
+   */
+  void addStandardNonlinearConvergence(const std::vector<PostprocessorName> & postprocessors,
```

Comment:
```suggestion
  void addNonlinearConvergence(const std::vector<PostprocessorName> & postprocessors,
```
"Standard" concerns me because this is a different "standard" convergence than that in the framework

Reply:
`addNonlinearConvergence` sounds like it's the only way to add a nonlinear Convergence object for a component. Maybe just `addMultiPostprocessorConvergence`?